controller/state.go

```diff
@@ -221,6 +221,15 @@ func (s *ksonnetAppStateManager) CompareAppState(app *v1alpha1.Application, revi
 		conditions = append(conditions, v1alpha1.ApplicationCondition{Type: v1alpha1.ApplicationConditionComparisonError, Message: err.Error()})
 	}
 
+	for _, liveObj := range controlledLiveObj {
+		if appLabelVal, ok := liveObj.GetLabels()[common.LabelApplicationName]; ok && appLabelVal != "" && appLabelVal != app.Name {
```

Comment:
This has potential for panic if GetLabels() returns nil, which might happen if we adopt an existing app. Can we guard against this?

Reply:
Good catch. `liveObj` also might be nil. Added missing check